test/unit/model/delivery/task/LtiDeliveryCreationTaskTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoLtiConsumer\test\unit\model\delivery\task;
+
+use common_exception_MissingParameter as MissingParameterException;
+use common_exception_InconsistentData as InconsistentDataException;
+use common_report_Report as Report;
+use core_kernel_classes_Class as RdfClass;
+use core_kernel_classes_Resource as RdfResource;
+use oat\generis\model\data\Model;
+use oat\generis\test\TestCase;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoLtiConsumer\model\delivery\factory\LtiDeliveryFactory;
+use oat\taoLtiConsumer\model\delivery\task\LtiDeliveryCreationTask;
+use PHPUnit_Framework_MockObject_MockObject as MockObject;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+class LtiDeliveryCreationTaskTest extends TestCase
+{
+    /**
+     * @dataProvider paramsToTest
+     *
+     * @param string $deliveryClassName
+     * @param bool   $existingClass
+     * @param string $label
+     * @param string $deliveryResourceId
+     *
+     * @throws MissingParameterException
+     * @throws InconsistentDataException
+     */
+    public function testInvoke($deliveryClassName, $existingClass, $label, $deliveryResourceId)
```

Comment:
maybe we can use setUp and helpers functions,
and by doing this we will decrease the number of lines in this fat function also will increase the readability

Reply:
Yes, I'd love to be able to do a `createConfiguredMock`, but it's not available on the phpunit version we're currently using (and lock because of php version...)
In fact, it wouldn't simplify a lot, since it can not check `with` in the same time, it's just `willReturn` configuration.